libraries/AP_Math/AP_Math.h

```diff
@@ -1,23 +1,15 @@
-// -*- tab-width: 4; Mode: C++; c-basic-offset: 4; indent-tabs-mode: nil -*-
+#pragma once
 
-#ifndef AP_MATH_H
-#define AP_MATH_H
+#include <limits>
+#include <type_traits>
+#include <cmath>
 
-// MATH_CHECK_INDEXES modifies some objects (e.g. SoloGimbalEKF) to
-// include more debug information.  It is also used by some functions
-// to add extra code for debugging purposes.  If you wish to activate
-// this, do it here or as part of the top-level Makefile -
-// e.g. Tools/Replay/Makefile
-#ifndef MATH_CHECK_INDEXES
-#define MATH_CHECK_INDEXES 0
-#endif
-
-// Assorted useful math operations for ArduPilot(Mega)
+#include <stdint.h>
 
-#include <AP_Common/AP_Common.h>
 #include <AP_Param/AP_Param.h>
-#include <math.h>
-#include <stdint.h>
+#include <AP_Common/AP_Common.h>
+
+#include "definitions.h"
```

Comment:
definitions.h is only included here which means that any header included previously will not see MATH_CHECK_INDEXES and thus give you warnings.